README_WAKE.adoc

```diff
@@ -0,0 +1,154 @@
+= Rocket Chip Generator
+
+This repository contains the Rocket chip generator necessary to instantiate
+the RISC-V Rocket Core. For more information on Rocket Chip, please consult our
+http://www.eecs.berkeley.edu/Pubs/TechRpts/2016/EECS-2016-17.html[technical report].
+
+== Dependencies
+
+This repository uses version 0.10.1 of https://github.com/sifive/wit[Wit] and version 0.15.1 of https://github.com/sifive/wake[Wake].
+Please see the respective repositories and their releases to install these dependencies.
+
+== Checkout The Code
+
+This repository uses https://github.com/sifive/wit[Wit] for dependency fetching and resolution.
+Please see the https://github.com/sifive/wit/blob/v0.10.1/share/doc/wit/tutorial.md[Wit tutorial] for more information.
+
+----
+$ wit init workspace -a git@github.com:chipsalliance/rocket-chip.git
+----
+
+This will clone rocket-chip as well as resolve and fetch all dependencies.
+
+== Compile Rocket Chip
+
+It is not necessary to compile separately from invocation, but it might be useful to understand how the different pieces work.
+
+This repository uses https://github.com/sifive/wake[Wake] for orchestrating the build.
+We strongly recommend checking out the https://github.com/sifive/wake/blob/v0.15.1/share/doc/wake/tutorial.md[Wake tutorial].
+You can find the Rocket Chip Wake rules in `build.wake` within this repository.
+
+The Rocket Chip Scala code is compiled using https://github.com/sifive/api-scala-sifive[API Scala SiFive],
+and can thus be compiled with:
+
+----
+$ wake compileScalaModule rocketchipScalaModule
+----
+
+The return value of that function is the `Path` to the compile jar.
+
+== Run Rocket Chip Generator
+
+=== API
+
+You can run the generator with the Wake rule `runRocketChipGenerator`.
+This function accepts a single `RocketChipGeneratorOptions` as an argument.
+
+----
+tuple RocketChipGeneratorOptions =
+  global Jars:          List Path
+  global TargetDir:     Path
+  global TopModuleName: String
+  global ConfigNames:   List String
+  global ExtraSources:  List Path
+  global BaseFileName:  Option String
+  
+global def makeRocketChipGeneratorOptions jars targetDir topModule configs =
+  RocketChipGeneratorOptions jars targetDir topModule configs Nil None
+----
+Please see the https://github.com/sifive/wake/blob/a0d99f15cf806f917d64714934d337b392d36f26/share/doc/wake/tour/tuples.adoc[Wake tuple documentation]
+for more information about Tuples.
+
+Because the constructor is private, we use `makeRocketChipGeneratorOptions` to construct `RocketChipGeneratorOptions`:
+
+* `jars` is a List of compiled Scala jars for your project, for example `rocketchipScalaModule.scalaModuleClasspath` for base Rocket Chip.
+* `targetDir` is the directory where the output of the generator will be placed.
+* `topModule` is the name of the top Chisel Module, eg. `"freechips.rocketchip.system.TestHarness"`
```

Comment:
Might be good to replace "name" with "fully-qualified name".

Reply:
Agreed